woocommerce/woocommerce-gutenberg-products-block
# Fix: load blocks style separately for classic themes

## What

Fixes #10546 

- Revert #10543
- Adjust the block style loading logic for classic theme.

For better review experience, you can review only https://github.com/woocommerce/woocommerce-blocks/pull/10758/files/8d0dddb4821ac4a44cf13ba37e1d2ef1699a1dcd..9ad4b0eb0e10f2a10fd9f9a006952203b5950f90.

## Why

Because we shouldn't load the unnecessary styles.


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

Testing this PR means making sure there are no styling regressions in existing blocks for classic themes.

0. Activate Storefront theme.
1. Create a post or page and add the All Products block. Verify styles are loaded correctly.
2. Visit the page in the frontend and verify styles are loaded correctly in the frontend as well.
3. Repeat steps 1 and 2 with all blocks listed on [this page](https://wordpress.org/plugins/woo-gutenberg-products-block/). Make sure to test each block individually. So, when possible, try with only one block on the page (in some cases, that's not possible, ie: filter blocks, in that case, try with as few blocks as possible on the page). The reason is that we want to make sure each block includes the style dependencies that it needs, so they need to be tested in isolation, otherwise styles from other blocks might leak into other blocks and "help fix issues".
4. Repeat steps above with a block themes. Block themes are excluded from the newly added logic so there shouldn't be any regression with blocks theme as well, but we should ,at least, smoke test with a block theme.


## Screenshots or screencast



N/a

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Fix: load blocks style separately for classic themes.
